File: schemaorg-rs/src/shipping_delivery_time.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///https://schema.org/additionalType
///https://schema.org/Text
///https://schema.org/URL
pub type ShippingDeliveryTimeAdditionalTypeFieldEnum = String;
///https://schema.org/ShippingDeliveryTime
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct ShippingDeliveryTime {
    #[serde(rename = "@context")]
    pub context: String,
    ///https://schema.org/additionalType
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<ShippingDeliveryTimeAdditionalTypeFieldEnum>,
    ///https://schema.org/transitTime
    #[serde(rename = "transitTime")]
    #[serde_as(as = "OneOrMany<_>")]
    pub transit_time: Vec<ShippingDeliveryTimeTransitTimeFieldEnum>,
    ///https://schema.org/subjectOf
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<ShippingDeliveryTimeSubjectOfFieldEnum>,
    ///https://schema.org/identifier
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<ShippingDeliveryTimeIdentifierFieldEnum>,
    ///https://schema.org/name
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///https://schema.org/disambiguatingDescription
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///https://schema.org/url
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///https://schema.org/description
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<ShippingDeliveryTimeDescriptionFieldEnum>,
    ///https://schema.org/image
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<ShippingDeliveryTimeImageFieldEnum>,
    ///https://schema.org/handlingTime
    #[serde(rename = "handlingTime")]
    #[serde_as(as = "OneOrMany<_>")]
    pub handling_time: Vec<ShippingDeliveryTimeHandlingTimeFieldEnum>,
    ///https://schema.org/cutoffTime
    #[serde(rename = "cutoffTime")]
    #[serde_as(as = "OneOrMany<_>")]
    pub cutoff_time: Vec<String>,
    ///https://schema.org/mainEntityOfPage
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<ShippingDeliveryTimeMainEntityOfPageFieldEnum>,
    ///https://schema.org/alternateName
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///https://schema.org/potentialAction
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///https://schema.org/sameAs
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///https://schema.org/businessDays
    #[serde(rename = "businessDays")]
    #[serde_as(as = "OneOrMany<_>")]
    pub business_days: Vec<ShippingDeliveryTimeBusinessDaysFieldEnum>,
}

File: schemaorg-ts/src/shipping_delivery_time.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type ShippingDeliveryTimeAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/ShippingDeliveryTime>
 **/
export interface ShippingDeliveryTime {
  "@context": string;
  "@type": "ShippingDeliveryTime";

  /**
   * <https://schema.org/businessDays>
   **/
  businessDays: all.OneOrMany<all.ShippingDeliveryTimeBusinessDaysFieldEnum>;

  /**
   * <https://schema.org/cutoffTime>
   **/
  cutoffTime: all.OneOrMany<string>;

  /**
   * <https://schema.org/handlingTime>
   **/
  handlingTime: all.OneOrMany<all.ShippingDeliveryTimeHandlingTimeFieldEnum>;

  /**
   * <https://schema.org/transitTime>
   **/
  transitTime: all.OneOrMany<all.ShippingDeliveryTimeTransitTimeFieldEnum>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.ShippingDeliveryTimeMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.ShippingDeliveryTimeIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.ShippingDeliveryTimeDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.ShippingDeliveryTimeSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.ShippingDeliveryTimeAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.ShippingDeliveryTimeImageFieldEnum>;
}
